
# HG changeset patch
# User Ozkan Sezer <sezeroz@gmail.com>
# Date 1541526624 -10800
# Node ID 244cdac21bcdd0e4a6152ad7c28756bcb39c4aa9
# Parent  50b071ef40ab427420dbe91212473e88c7d42155
fix bug #4362 - SDL_syswm.h with SDL_PROTOTYPES_ONLY broken in C++ mode

diff -r 50b071ef40ab -r 244cdac21bcd include/SDL_syswm.h
--- a/include/SDL_syswm.h	Sun Nov 04 21:11:07 2018 -0500
+++ b/include/SDL_syswm.h	Tue Nov 06 20:50:24 2018 +0300
@@ -41,9 +41,9 @@
  *  an unhandled window event occurs.  This event is ignored by default, but
  *  you can enable it with SDL_EventState().
  */
-#ifdef SDL_PROTOTYPES_ONLY
 struct SDL_SysWMinfo;
-#else
+
+#if !defined(SDL_PROTOTYPES_ONLY)
 
 #if defined(SDL_VIDEO_DRIVER_WINDOWS)
 #ifndef WIN32_LEAN_AND_MEAN
@@ -103,6 +103,8 @@
 #if defined(SDL_VIDEO_DRIVER_VIVANTE)
 #include "SDL_egl.h"
 #endif
+#endif /* SDL_PROTOTYPES_ONLY */
+
 
 #include "begin_code.h"
 /* Set up for C function definitions, even when using C++ */
@@ -110,6 +112,7 @@
 extern "C" {
 #endif
 
+#if !defined(SDL_PROTOTYPES_ONLY)
 /**
  *  These are the various supported windowing subsystems
  */


